src/docker_clojure/core.clj

```diff
@@ -0,0 +1,114 @@
+(ns docker-clojure.core
+  (:require
+   [clojure.java.shell :refer [sh]]
+   [clojure.math.combinatorics :as combo]
+   [clojure.string :as str]
+   [docker-clojure.dockerfile :as df]))
+
+(def base-images
+  #{"openjdk:8" "openjdk:11"})
+
+(def distros
+  #{"debian" "alpine"})
+
+(def build-tools
+  {"lein"       "2.8.1"
+   "boot"       "2.8.1"
+   "tools-deps" "1.9.0.397"})
+
+(def exclusions ; don't build these for whatever reason(s)
+  #{{:base-image "openjdk:11"
+     :distro     "alpine"}})
+
+(def maintainers
+  {:paul "Paul Lam <paul@quantisan.com>"
+   :wes  "Wes Morgan <wesmorgan@icloud.com>"
+   :dlg  "Kirill Chernyshov <delaguardo@gmail.com>"})
+
+(defn maintainer [{:keys [distro build-tool]}]
+  (cond
+    (and (= distro "debian") (= build-tool "lein"))
+    (:paul maintainers)
+
+    (and (= distro "debian") (= build-tool "boot"))
+    (:wes maintainers)
+
+    (= build-tool "tools-deps")
+    (:dlg maintainers)
+
+    (= distro "alpine")
+    (:wes maintainers)))
+
+(defn exclude? [exclusions variant]
+  (some (fn [exclusion]
+          (every? (fn [[k v]]
+                    (= v (get variant k)))
+                  exclusion))
```

Comment:
code smell. perhaps tweak the `exclusions` var and/or `variant` map, or overwrite `equals` fn on `defrecord variant` so that you can ultimately do something like 

```
(defn exclude [exclusions variant ]
  (contains? (into #{} exclusions) variant))
```

Reply:
What specifically is the code smell here?

I tend to prefer basic data structures over records and methods.

The logic I’m trying to achieve is that exclusions don’t have to be comprehensive, but every key-value they _do_ specify must all be matched in a variant for it to be excluded. I don’t really love hiding that logic.

Reply:
perhaps add that as a code comment? the logic isn't obvious on first glance, thus I consider that as code smell.

and maybe refactor out an `contains-every-key-value?` helper fn in an `letfn`. so that you can do,

```
(some (partial contains-every-key-value? variant) exclusion)
```

Reply:
OK, makes sense. I’ll work on refactoring it a bit to make it clearer what it’s doing and add comments if anything still seems unintuitive.